hal/halx86/mp/ioapic.c

```diff
@@ -647,13 +648,14 @@ HaliReconfigurePciInterrupts(VOID)
 	        i, IRQMap[i].IrqType, IRQMap[i].IrqFlag, IRQMap[i].SrcBusId,
 	        IRQMap[i].SrcBusIrq, IRQMap[i].DstApicId, IRQMap[i].DstApicInt);
 
-	 HalSetBusDataByOffset(PCIConfiguration,
-	                               IRQMap[i].SrcBusId,
-				       (IRQMap[i].SrcBusIrq >> 2) & 0x1f,
-				       &IRQMap[i].DstApicInt,
-				       0x3c /*PCI_INTERRUPT_LINE*/,
-				       1);
-
+         Size = HalSetBusDataByOffset(PCIConfiguration,
+                                      IRQMap[i].SrcBusId,
+                                      (IRQMap[i].SrcBusIrq >> 2) & 0x1f,
+                                      &IRQMap[i].DstApicInt,
+                                      0x3c /* PCI_INTERRUPT_LINE */,
+                                      1);
+         if (Size == 0)
```

Comment:
I wonder whether a comparison with the written size would be better.... `if (Size != sizeof(UCHAR))`